x-pack/plugins/triggers_actions_ui/public/application/lib/validate_params_for_warnings.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { intersection, get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+
+const publicUrlWarning = i18n.translate(
+  'xpack.triggersActionsUI.sections.actionTypeForm.warning.publicUrl',
+  {
+    defaultMessage: 'Kibana missing publicUrl environment variable. Action will use relative URLs.',
+  }
+);
+
+export function validateParamsForWarnings(
+  actionParams: Record<string, unknown>,
+  publicBaseUrl: string | undefined
+): {
+  warnings: Record<string, any>;
+} {
+  const publicUrlFields = ['context.alertDetailsUrl', 'context.viewInAppUrl'];
```

Comment:
What do you think of updating the `ActionVariable` type from `x-pack/plugins/alerting/common/rule.ts` to include a flag like `usesPublicBaseUrl`? That way we can allow new action variables to define whether the warning would get shown for them vs maintaining an allowlist of fields? Note that `rule.url` is another context variable that uses the public base url.

Reply:
oh yeah, I like that idea!

Reply:
If you did that, you could just update the logic inside the `TextAreaWithMessageVariables` component to look at the action variables that are passed in to determine whether to show the warning.

Reply:
I think we only want to show that warning if the user uses on of those variables though

Reply:
Here is what I was thinking, https://github.com/elastic/kibana/pull/149832/commits/9290e647a0a6ae1fbb3167d25dc976f28681f974